packages/core/src/browser/fetchObservable.ts

```diff
@@ -96,22 +97,19 @@ function afterSend(
   const reportFetch = (response: Response | Error) => {
     const context = startContext as unknown as FetchCompleteContext
     context.state = 'complete'
-    context.duration = elapsed(context.startClocks.timeStamp, timeStampNow())
-
+    context.resolveDuration = elapsed(startContext.startClocks.timeStamp, timeStampNow())
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Shouldn't it be in `FetchCompleteContext` instead of the `duration` attribute?

Reply:
I am not sure I follow, do you mean to say that the variable should probably be defined in `beforeSend` where we define the `FetchCompleteContext`?

If so, I do not think this is the right place to put it. 

`reportFetch` gets called only after the `fetch` `.then` has been triggered. Moving this logic to `FetchCompleteContext` (aka `beforeSend`) would mean we incorrectly calculating `resolveDuration`